doc/src/guides/solving/solve-numerically.md

```diff
@@ -0,0 +1,207 @@
+
+## Guidance
+
+### Find complex roots of real function
+
+To solve for complex roots of real functions, specify a nonreal (either purely imaginary, or complex) initial point:
+
+```py
+>>> from sympy import nsolve
+>>> from sympy.abc import x
+>>> nsolve(x**2 + 2, 1) # Real initial point returns no root
+Traceback (most recent call last):
+    ...
+ValueError: Could not find root within given tolerance. (4.18466446988997098217 > 2.16840434497100886801e-19)
+Try another starting point or tweak arguments.
+>>> from sympy import I
+>>> nsolve(x**2 + 2, I) # Imaginary initial point returns a complex root
+1.4142135623731*I
+>>> nsolve(x**2 + 2, 1 + I) # Complex initial point returns a complex root
+1.4142135623731*I
+```
+
+### Ensure the root found is in a given interval
+
+It is not guaranteed that nsolve will find the root closest to the initial point. Here, even though the root `-1` is closer to the initial point of `-0.1`, nsolve finds the root `1`:
+
+```py
+>>> from sympy import nsolve
+>>> from sympy.abc import x
+>>> nsolve(x**2 - 1, -0.1)
+1.00000000000000
+```
+
+You can ensure the root found is in a given interval, if such a root exists, using `solver='bisect'` by specifying the interval in a tuple. Here, specifying the interval `(-10, 0)` ensures that the root `-1` is found:
+
+```py
+>>> from sympy import nsolve
+>>> from sympy.abc import x
+>>> nsolve(x**2 - 1, (-10, 0), solver='bisect', verify=False)
+-1.00000000000000
+```
+
+## Solve a system of equations numerically
+
+### Solve multidimensional functions
+
+To solve multidimensional functions, supply a tuple of
+- functions `(f1, f2)`
+- variables to solve for `(x1, x2)`
+- starting values `(-1, 1)`
+
+```py
+>>> from sympy import Symbol, nsolve
+>>> x1 = Symbol('x1')
+>>> x2 = Symbol('x2')
+>>> f1 = 3 * x1**2 - 2 * x2**2 - 1
+>>> f2 = x1**2 - 2 * x1 + x2**2 + 2 * x2 - 8
+>>> print(nsolve((f1, f2), (x1, x2), (-1, 1)))
+Matrix([[-1.19287309935246], [1.27844411169911]])
+```
+
+### Solve overdetermined systems
+
+In an overdetermined system, there are more equations than unknowns. In general, this means that there is no exact solution, but there will be an approximate solution that is the closest fit to the equations. Overdetermined systems are common when the data is noisy, and the approximate solution is useful rather than giving up because there is no exact solution.
+
+*Thought this would work but it throws an error*
+
+from sympy import Symbol, nsolve
+
+x1 = Symbol('x1')
+
+x2 = Symbol('x2')
+
+f1 = 3 * x1**2 - 2 * x2**2 - 1
+
+f2 = x1**2 - 2 * x1 + x2**2 + 2 * x2 - 8
+
+f3 = x1 + x2
+
+print(nsolve((f1, f2, f3), (x1, x2), (-1, 1)))
+
+### Use SciPy on a lambda function *what is the advantage of this method?*
+
+You can *description*
+
+```py
+>>> from sympy.abc import x
+>>> from sympy.utilities.lambdify import implemented_function
+>>> from sympy import lambdify
+>>> from scipy import optimize
```

Comment:
This doctest failed because SciPy wasn't available for import. How do I add that module? Or is using SciPy a bad idea?

Reply:
You need to add it to `_get_doctest_blacklist` in `sympy/testing/runtests.py` so that it only gets run when scipy is installed.

Reply:
I tried to follow the pattern there, if this isn't the correct technique please let me know:

```
    if import_module("scipy") is None:
        # throws ModuleNotFoundError when scipy not installed
        blacklist.extend(
            ["doc/src/guides/solving/solve-numerically.md",]
        )
```

And should I add scipy to `doc/requirements.txt` so scipy will be available for import? I would think we don't want to permanently prevent doctests on the new page `doc/src/guides/solving/solve-numerically.md`...

Reply:
> And should I add scipy to doc/requirements.txt so scipy will be available for import? 

No that file is just for things needed to build the documentation. 

> I would think we don't want to permanently prevent doctests on the new page doc/src/guides/solving/solve-numerically.md...

If you did the logic correctly it will skip the file when scipy isn't installed but run the doctests when it is. The optional dependencies build on CI has scipy installed so you can check if it is running the tests there.

Reply:
Understand. I checked tests such as `doctests (3.8, false)` and it includes the [new page passing doctest](https://github.com/sympy/sympy/runs/7587802527?check_suite_focus=true#step:6:531), so seems like my addition to `_get_doctest_blacklist` in `sympy/testing/runtests.py`.